src/humanloop/client.py

```diff
@@ -78,11 +107,304 @@ def __init__(
+        the [Tool](https://humanloop.com/docs/explanation/tools).
+
+        If the [Tool](https://humanloop.com/docs/explanation/tools) already exists
+        on the specified path, a new version will be upserted when any of the
+        above change.
+
+        Here's an example of declaring a [Tool](https://humanloop.com/docs/explanation/tools) in code:
+
+        ```python
+        @tool
+        def calculator(a: int, b: Optional[int]) -> int:
+            \"\"\"Add two numbers together.\"\"\"
+            return a + b
+        ```
+
+        This will create a [Tool](https://humanloop.com/docs/explanation/tools) with the following attributes:
+        ```python
+        {
+            strict: True,
+            function: {
+                "name": "calculator",
+                "description": "Add two numbers together.",
+                "parameters": {
+                    type: "object",
+                    properties: {
+                        a: {type: "integer"},
+                        b: {type: "integer"}
+                    },
+                    required: ["a"],
+                },
+            }
+        }
+        ```
+
+        Every call to the decorated function will create a Log against the Tool. For example:
+
+        ```python
+        calculator(a=1, b=2)
+        ```
+
+        Will create the following Log:
+
+        ```python
+        {
+            "inputs": {
+                a: 1,
+                b: 2
+            },
+            "output": 3
+        }
+
+        :param path: The path to the Tool. If not provided, the function name
+            will be used as the path and the File will be created in the root
+            of your Humanloop's organization workspace.
+
+        :param setup_values: Values needed to setup the Tool, defined in
+            JSON Schema format: https://json-schema.org/
+
+        :param attributes: Additional fields to describe the Tool.
+            Helpful to separate Tool versions from each other
+            with details on how they were created or used.
+        """
+        return tool_decorator_factory(
+            path=path,
+            setup_values=setup_values,
+            attributes=attributes,
+        )
+
+    def flow(
+        self,
+        *,
+        path: Optional[str] = None,
+        attributes: Optional[dict[str, typing.Any]] = None,
+    ):
+        """Decorator for declaring a [Flow](https://humanloop.com/docs/explanation/flows) in code.
+
+        A [Flow](https://humanloop.com/docs/explanation/flows) decorator should be added
+        at the entrypoint of your LLM feature. Call other functions decorated with
+        Humanloop SDK decorators to create a Trace of Logs on Humanloop.
+
+        Here's an example of declaring a [Flow](https://humanloop.com/docs/explanation/flows) in code:
+        ```python
+        @prompt(template="You are an assistant on the following topics: {topics}.")
```

Comment:
this is a prompt example :( please go through and fix up any docstring issues

Reply:
That was intentional: I wanted to showcase that Flows can be used in tandem with other File decorators to create a Flow Trace. Here's an amended docstring that clarifies.

<img width="809" alt="image" src="https://github.com/user-attachments/assets/242df5b1-a72c-430d-8254-6b06603e0f65">